### PR TITLE
Analysis/ job errors

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Added job errors page to display galaxy job error information

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

**Comment by joshsadam** [comment]
Can you organize your components into directories so I know which ones go together?

**Comment by deepsidhu85** [comment]
> Can you organize your components into directories so I know which ones go together?

Done!

File: pom.xml [hunk 1]
```diff
@@ -623,6 +623,11 @@
 			<artifactId>jvm-breakglass</artifactId>
 			<version>${jvm-breakglass.version}</version>
 		</dependency>
+		<dependency>
+			<groupId>com.google.code.gson</groupId>
+			<artifactId>gson</artifactId>
+			<version>2.8.6</version>
```
Comment (joshsadam) [inline pom.xml]: Why did you add this?

File: src/main/webapp/resources/js/components/typography/CodeBlock.jsx [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
+
+export function CodeBlock({ children }) {
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/typography/CodeBlock.jsx]: Add a comment here on when to use etc...

File: src/main/webapp/resources/js/components/typography/Monospace.jsx [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
+
+export function Monospace({ children }) {
```
Comment (joshsadam) [inline src/main/webapp/resources/js/components/typography/Monospace.jsx]: Comment

File: src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx [hunk 4]
```diff
@@ -45,6 +45,12 @@ const analysisTypesWithAdditionalPage = [
...
 
 const analysisSettingsTabKeys = ["details", "samples", "share", "delete"];
+const analysisErrorTabKeys = [
+  "job-error-info",
+  "galaxy-parameters",
+  "standard-error",
+  "standard-out"
+];
 
 const Wrapper = styled.div`
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx]: `Wrapper` is not used anymore

File: src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx [hunk 5]
```diff
@@ -45,6 +45,12 @@ const analysisTypesWithAdditionalPage = [
...
   margin-left: ${SPACE_MD};
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx]: On line 62, `analysisContextUpdateSubmissionName` is not used.

File: src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx [hunk 6]
```diff
@@ -1,12 +1,140 @@
...
+const StandardOutputTab = React.lazy(() => import("./StandardOutputTab"));
+
+const TabPane = Tabs.TabPane;
+
+export default function AnalysisError(props) {
+  const { analysisContext } = useContext(AnalysisContext);
+  const [jobErrors, setJobErrors] = useState(null);
+  const [currActiveKey, setCurrActiveKey] = useState(1);
+
+  const StyledTabs = styled(Tabs)`
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx]: Can you, or did you, make this a proper component?

File: src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx [hunk 7]
```diff
@@ -1,12 +1,140 @@
...
+  `;
+
+  // Sets the job errors into a local state variable on page load
+  useEffect(() => {
+    getJobErrors(analysisContext.analysis.identifier).then(data => {
+      setJobErrors(data);
+    });
+  }, []);
+
+  // Sets the current activekey for the 'Pass' tabs
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx]: Change `activekey` to `active key`

File: src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx [hunk 8]
```diff
@@ -1,12 +1,140 @@
...
+  function updateActiveKey(key) {
+    setCurrActiveKey(key.charAt(key.length - 1));
+  }
 
-export default function AnalysisError() {
   return (
-    <Title level={2} className="t-page-title">
-      Job Error
-    </Title>
+    <>
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx]: Try to break this section into components, you also don't need the `<>`

File: src/main/webapp/resources/js/pages/analysis/components/GalaxyJobInfoTab.jsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { PassTabs } from "./PassTabs";
+import { GalaxyJobInfo } from "./GalaxyJobInfo";
+import { Col, Typography } from "antd";
+import { getI18N } from "../../../utilities/i18n-utilties";
+
+const { Title } = Typography;
+
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/analysis/components/GalaxyJobInfoTab.jsx]: Make sure you have comments on the top of files says how or why they are used.

File: src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx [hunk 10]
```diff
@@ -1,12 +1,365 @@
...
+    });
+  }, []);
+
+  // Returns the galaxy job details for the given index
+  function galaxyJobDetails(index) {
+    return([
+      {
+        title: getI18N("AnalysisError.createdDate"),
+        desc: (
+          <Text>
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx]: Already done.

File: src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx [hunk 11]
```diff
@@ -1,12 +1,365 @@
...
+            {formatDate({
+              date:
+                jobErrors.galaxyJobErrors[index]
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx]: Already done.

File: pom.xml (hunk 1 above)
Comment (deepsidhu85) [reply pom.xml]: Needed the JSON parser out of it since the `parameters` string is not in valid json. The parser helps convert the parameters to a valid json string before returning

File: src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx (hunk 5 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx]: Alread done.

File: src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx (hunk 6 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx]: I have in a branch that is based off this one.

File: src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx (hunk 4 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx]: Already done.

File: src/main/java/ca/corefacility/bioinformatics/irida/model/workflow/analysis/JobError.java [hunk 12]
```diff
@@ -292,70 +294,50 @@ public boolean equals(Object o) {
 		if (o == null || getClass() != o.getClass())
 			return false;
 		JobError jobError = (JobError) o;
-		return getExitCode() == jobError.getExitCode() &&
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/model/workflow/analysis/JobError.java]: Weird. This formatting was done in Intellij with our formatting rules for java files

[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/model/workflow/analysis/JobError.java]: You can have you java format only CVS changed code.

File: src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx (hunk 8 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx]: Already done. However removing the <> breaks the page and there are a bunch of errors

File: src/main/webapp/resources/js/pages/analysis/components/GalaxyJobInfoTab.jsx (hunk 9 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/GalaxyJobInfoTab.jsx]: Added in 39c89d161e41dc4241d4d58b6333be75bf9d2a92

File: src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx (hunk 7 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx]: Fixed in 39c89d161e41dc4241d4d58b6333be75bf9d2a92

File: src/main/webapp/resources/js/components/typography/CodeBlock.jsx (hunk 2 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/components/typography/CodeBlock.jsx]: Added in 39c89d161e41dc4241d4d58b6333be75bf9d2a92

File: src/main/java/ca/corefacility/bioinformatics/irida/model/workflow/analysis/JobError.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/model/workflow/analysis/JobError.java]: Fixed in 39c89d161e41dc4241d4d58b6333be75bf9d2a92

File: src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx (hunk 8 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/AnalysisError.jsx]: Fixed in 6e887213d951c73e40a422689baebcb640ab1be4
